bezier-rs/lib/Cargo.toml

```diff
@@ -12,3 +12,4 @@ license = "Apache-2.0"
 
 [dependencies]
 glam = { version = "0.17", features = ["serde"] }
+derive_builder = "0.11.2"
```

Comment:
I am open to being convinced otherwise, but I don't feel like the builder pattern is quite right for our needs (it seems to be generally the right idea to use another struct, but overkill to have this infrastructure surrounding the options struct). How about just defining the options struct (which derives `Default`) and just manually building it the normal way (like below)?

```rust
FooOptions::default()
```
or:
```rust
FooOptions {
    quick: 123,
    brown: 456,
    fox: 789,
    ..Default::default()
}
```

Reply:
That sounds good to me, but I'm no expert with Rust. @TrueDoctor @otdavies Do you two have any additional thoughts on this? I also might not be using the builder pattern correctly, so I'd appreciate your thoughts on this!

Reply:
One advantage is that you don't have to provide your own default implementation by using the struct annotations.
The second advantage is that this works for fileds that don't implement default.
You can for example have a mix of required fields and optional fields which wouldn't be the case with the struct initializer.
You could of course provide your own default struct 
```rust
FooOptions {
 quick: 123,
 brown: 456,
 fox: 789,
  ..MyDefauilt }
```
but that would require the field types to implement Clone.
In this singular example the advantage of using the builder_derive crate is negligable but I thought about introducing this dep anyway because it might be useful in other areas of the project.

Reply:
@Keavon based on the additional comment, do you still prefer the struct implementation over builder? I don't have a particular preference as I agree that builder isn't super necessary here, but it may be nice to include the dependency now if we might use it in the future.

Reply:
Since this is a library used by others, is especially important to keep down on the dependencies. I think that need outweighs the slight benefits @TrueDoctor mentioned (but thanks for describing those!). Let's stick to the simple option and just implement default.